common/reviews/api/node-core-library.api.md

```diff
@@ -33,7 +33,9 @@ export class AnsiEscape {
 
 // @beta
 export class Async {
-    static forEachAsync<TEntry>(iterable: Iterable<TEntry> | AsyncIterable<TEntry>, callback: (entry: TEntry, arrayIndex: number) => Promise<void>, options?: IAsyncParallelismOptions | undefined): Promise<void>;
+    static forEachAsync<TEntry>(iterable: Iterable<TEntry> | AsyncIterable<TEntry>, callback: (entry: TEntry, arrayIndex: number) => Promise<void>, options?: TEntry extends {
+        weight?: number;
```

Comment:
This type algebra seems overcomplicated. I think is trying to say that IF the options specifies `useWeight`, THEN the entry must define a `weight`.  I don't think it actually accomplishes that, but it is a slightly odd requirement given that both these properties are optional.  If weight defaults to `1`, then is it really important to have been able to specify it, even though we chose not to?

I think it might be better to simply say that "If `useWeight`=true, then `forEachSync` will use the `weight` property if present, otherwise the `weight` defaults to `1`."  The only realistic concern here is if old code randomly had a `weight` property, that suddenly gains new meaning in this release, but you've already solved that by defaulting `useWeight` to false.

Reply:
The check on the optional property is to ensure that the `TEntry` type does not have a property `weight` that is _incompatible_ with type `number`. I originally considered writing this as an overload, but I'm generally leery of overloads in case something uses `typeof Async.forEachAsync` somewhere. Might be cleanest to just add a completely separate entry point.